src/js/common/components/CampaignShare/ShareByCopyLink.jsx

```diff
@@ -96,6 +97,23 @@ class ShareByCopyLink extends Component {
     if (this.props.onClickFunction) {
       this.props.onClickFunction();
     }
+    // adding dataLayer for "Copy link" on Share Ballot
+    const currentPathname = window.location.pathname;
+
+    const dataLayerObject = {
+      event: 'ShareBallotCopyLinkClick',
+      shareDetails: {
+        shareType: this.props.shareType || 'ballotWithChoices',
+        campaignXWeVoteId: this.props.campaignXWeVoteId || null,
+      },
+      pageDetails: {
+        pageName: 'ShareBallotModal',
```

Comment:
To find the attributes for pageDetails, please use:
    const page = lookupPageNameAndPageTypeDict(currentPathname);

...and then:
```
        pageDetails: {
          pageName: page.pageName,
          pageType: page.pageType,
          pathname: currentPathname,
        },
```
You will need to update lookupPageNameAndPageTypeDict with the paths of the most common pages where ShareByCopyLink is called.